src/charts/dialogs/AddViewDialog.tsx

```
@@ -18,22 +18,27 @@ const AddViewDialogComponent = (props: {
+    //we want a choice of all datasources (not just ones on the current view)
+    //not sure how to get all the datasources available - this is just a temporary hack
+    //if we don't use useMemo get constant re-rendering ?
+    const dataSources = useMemo(()=>viewManager.getAllDataSources(),[viewManager])
+    const currentDataSources = useMemo(() => Object.keys(viewData.dataSources), [viewData.dataSources]);
 
     useEffect(() => {
         // Initialising the checkedDs with the data sources inside the view
         const tempDs: { [name: string]: boolean } = {};
         dataSources?.forEach?.((ds) => {
-            tempDs[ds] = false;
+            tempDs[ds] =currentDataSources.indexOf(ds) !=-1;
```

Biome (1.9.4)

[error] Use !== instead of !=. != is only allowed when comparing against `null`

!= is only allowed when comparing against null

Using != may be unsafe if you are relying on type coercion
Unsafe fix: Use !==


(lint/suspicious/noDoubleEquals)